### PR TITLE
OOC notes rewrite and examine boxes in HUD's output

### PR DESCRIPTION
**Pull request by SidVeld**
# OOC notes rewrite and examine boxes in HUD's output

This pull request contains two changes.

## OCC Notes: return

Gives you the ability to view OOC notes when examining a character. Requires `ALLOW_METADATA` to be enabled in the config.

изображение

## Examine boxes in Huds

I liked the examine boxes so much I wanted to use them somewhere else. I created three new subclasses of examine boxes and use them in med/sec hud's messages. I hope you enjoy it.

### Security

изображение

### Medical

изображение

### CODE REVIEW

**Comment by SleepyGemmy** [comment]
!review

File: code/modules/client/preference_setup/general/01_basic.dm [hunk 1]
```diff
@@ -302,10 +303,18 @@
 		return TOPIC_REFRESH
 
 	else if(href_list["metadata"])
-		var/new_metadata = sanitize(input(user, "Enter any information you'd like others to see, such as Roleplay-preferences:", "Game Preference" , pref.metadata) as message|null)
+		var/new_metadata = sanitize(
+			input(
+				user,
+				"Enter any information you'd like others to see, such as Roleplay-preferences:",
```
Comment (SleepyGemmy) [inline code/modules/client/preference_setup/general/01_basic.dm:309]:
```suggestion
				"Enter any information you'd like others to see, such as roleplay preferences.",
```

File: code/modules/client/preference_setup/general/01_basic.dm [hunk 2]
```diff
@@ -348,4 +357,20 @@
 			pref.machine_ownership_status = new_ownership_status
 			return TOPIC_REFRESH
 
+	else if (href_list["clear_metadata"])
+		if (CanUseTopic(user))
+			var/user_choice = alert(
+				user,
+				"Are you sure you wish to clear ooc notes of this character?",
+				"Clear OOC notes Confirmation",
```
Comment (SleepyGemmy) [inline code/modules/client/preference_setup/general/01_basic.dm:365]:
```suggestion
				"Are you sure you wish to clear this character's OOC notes?",
				"Clear OOC Notes Confirmation",
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 3]
```diff
@@ -576,7 +576,9 @@
 									U.handle_regular_hud_updates()
 
 			if(!modified)
-				to_chat(usr, SPAN_WARNING("Unable to locate a data core entry for this person."))
+				to_chat(
+					usr, EXAMINE_BLOCK_RED(SPAN_WARNING("Unable to locate a data core entry for this person."))
+				)
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:581]:
```suggestion
				to_chat(usr, EXAMINE_BLOCK_RED(SPAN_WARNING("Unable to locate a data core entry for this person.")))
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 4]
```diff
@@ -591,15 +593,18 @@
...
+						+ "<b>Notes:</b> [R.security.notes]\n" \
+						+ "<a href='?src=\ref[src];secrecordComment=`'>\[View Comment Log\]</a>"
+					to_chat(usr, EXAMINE_BLOCK_RED(message))
 					read = 1
 
 			if(!read)
-				to_chat(usr, SPAN_WARNING("Unable to locate a data core entry for this person."))
+				to_chat(
+					usr, EXAMINE_BLOCK_RED(SPAN_WARNING("Unable to locate a data core entry for this person."))
+				)
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:607]:
```suggestion
				to_chat(usr, EXAMINE_BLOCK_RED(SPAN_WARNING("Unable to locate a data core entry for this person.")))
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 5]
```diff
@@ -614,16 +619,20 @@
...
-					to_chat(usr, "<a href='?src=\ref[src];secrecordadd=`'>\[Add comment\]</a>")
+						message += "No comments found.\n"
+					message += "<a href='?src=\ref[src];secrecordadd=`'>\[Add comment\]</a>"
+					to_chat(usr, EXAMINE_BLOCK_RED(message))
 
 			if(!read)
-				to_chat(usr, SPAN_WARNING("Unable to locate a data core entry for this person."))
+				to_chat(
+					usr, EXAMINE_BLOCK_RED(SPAN_WARNING("Unable to locate a data core entry for this person."))
+				)
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:635]:
```suggestion
				to_chat(usr, EXAMINE_BLOCK_RED(SPAN_WARNING("Unable to locate a data core entry for this person.")))
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 6]
```diff
@@ -675,7 +684,9 @@
 								U.handle_regular_hud_updates()
 
 			if(!modified)
-				to_chat(usr, SPAN_WARNING("Unable to locate a data core entry for this person."))
+				to_chat(
+					usr, EXAMINE_BLOCK_DEEP_CYAN(SPAN_WARNING("Unable to locate a data core entry for this person."))
+				)
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:689]:
```suggestion
				to_chat(usr, EXAMINE_BLOCK_DEEP_CYAN(SPAN_WARNING("Unable to locate a data core entry for this person.")))
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 7]
```diff
@@ -690,15 +701,19 @@
...
+						+ "<b>Notes:</b> [R.medical.notes]\n" \
+						+ "<a href='?src=\ref[src];medrecordComment=`'>\[View Comment Log\]</a>"
+					to_chat(usr, EXAMINE_BLOCK_DEEP_CYAN(message))
 					read = 1
 
 			if(!read)
-				to_chat(usr, SPAN_WARNING("Unable to locate a data core entry for this person."))
+				to_chat(
+					usr, EXAMINE_BLOCK_DEEP_CYAN(SPAN_WARNING("Unable to locate a data core entry for this person."))
+				)
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:716]:
```suggestion
				to_chat(usr, EXAMINE_BLOCK_DEEP_CYAN(SPAN_WARNING("Unable to locate a data core entry for this person.")))
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 8]
```diff
@@ -713,16 +728,20 @@
 			var/datum/record/general/R = SSrecords.find_record("name", perpname)
 			if(istype(R) && istype(R.medical))
 				if(hasHUD(usr, "medical"))
+					var/message = "<b>Medical comments: [name]</b>\n\n"
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:731]:
```suggestion
					var/message = "<b>Medical Record Comments: [name]</b>\n\n"
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 9]
```diff
@@ -614,16 +619,20 @@
...
 					read = 1
 					if(R.security.comments.len > 0)
 						for(var/comment in R.security.comments)
-							to_chat(usr, comment)
+							message += comment + "\n\n"
 					else
-						to_chat(usr, "No comments found")
-					to_chat(usr, "<a href='?src=\ref[src];secrecordadd=`'>\[Add comment\]</a>")
+						message += "No comments found.\n"
+					message += "<a href='?src=\ref[src];secrecordadd=`'>\[Add comment\]</a>"
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:629]:
```suggestion
					message += "<a href='?src=\ref[src];secrecordadd=`'>\[Add Comment\]</a>"
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 10]
```diff
@@ -713,16 +728,20 @@
...
 					read = 1
 					if(R.medical.comments.len > 0)
 						for(var/comment in R.medical.comments)
-							to_chat(usr, comment)
+							message += comment + "\n\n"
 					else
-						to_chat(usr, "No comments found")
-					to_chat(usr, "<a href='?src=\ref[src];medrecordadd=`'>\[Add comment\]</a>")
+						message += "No comments found.\n"
+					message += "<a href='?src=\ref[src];medrecordadd=`'>\[Add comment\]</a>"
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:738]:
```suggestion
					message += "<a href='?src=\ref[src];medrecordadd=`'>\[Add Comment\]</a>"
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 11]
```diff
@@ -713,16 +728,20 @@
...
+					to_chat(usr, EXAMINE_BLOCK_DEEP_CYAN(message))
 
 			if(!read)
-				to_chat(usr, SPAN_WARNING("Unable to locate a data core entry for this person."))
+				to_chat(
+					usr, EXAMINE_BLOCK_DEEP_CYAN(SPAN_WARNING("Unable to locate a data core entry for this person."))
+				)
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:744]:
```suggestion
				to_chat(usr, EXAMINE_BLOCK_DEEP_CYAN(SPAN_WARNING("Unable to locate a data core entry for this person.")))
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 12]
```diff
@@ -791,6 +810,15 @@
 				flavor_texts[href_list["flavor_change"]] = msg
 				set_flavor()
 				return
+
+	if (href_list["metadata"])
+		var/message = "<b>OOC notes: [name]</b>" \
+			+ "\n\n" \
+			+ client.prefs.metadata \
+			+ "\n\n" \
+			+ SPAN_WARNING("Remember, this is ooc information.")
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:819]:
```suggestion
		var/message = "<b>OOC Notes: [name]</b>" \
			+ "\n\n" \
			+ client.prefs.metadata \
			+ "\n\n" \
			+ SPAN_WARNING("Remember, this is OOC information.")
```

File: code/modules/mob/living/carbon/human/examine.dm [hunk 13]
```diff
@@ -390,6 +390,9 @@
 
 	if(print_flavor_text()) msg += "[print_flavor_text()]\n"
 
+	if (GLOB.config.allow_Metadata && client.prefs.metadata)
+		msg += "<span class='deadsay'>OOC notes:</span> <a href='?src=\ref[src];metadata=1'>\[View\]</a>\n"
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/examine.dm:394]:
```suggestion
		msg += "<span class='deadsay'>OOC Notes:</span> <a href='?src=\ref[src];metadata=1'>\[View\]</a>\n"
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 14]
```diff
@@ -591,15 +593,18 @@
 			var/datum/record/general/R = SSrecords.find_record("name", perpname)
 			if(istype(R) && istype(R.security))
 				if(hasHUD(usr,"security"))
-					to_chat(usr, "<b>Name:</b> [R.name]")
-					to_chat(usr, "<b>Criminal Status:</b> [R.security.criminal]")
-					to_chat(usr, "<b>Crimes:</b> [R.security.crimes]")
-					to_chat(usr, "<b>Notes:</b> [R.security.notes]")
-					to_chat(usr, "<a href='?src=\ref[src];secrecordComment=`'>\[View Comment Log\]</a>")
+					var/message = "<b>Information</b>\n\n" \
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:596]:
```suggestion
					var/message = "<b>Security Records: [R.name]</b>\n\n" \
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 15]
```diff
@@ -690,15 +701,19 @@
 			var/datum/record/general/R = SSrecords.find_record("name", perpname)
 			if(istype(R) && istype(R.medical))
 				if(hasHUD(usr, "medical"))
-					to_chat(usr, "<b>Name:</b> [R.name]	<b>Blood Type:</b> [R.medical.blood_type]")
-					to_chat(usr, "<b>DNA:</b> [R.medical.blood_dna]")
-					to_chat(usr, "<b>Disabilities:</b> [R.medical.disabilities]")
-					to_chat(usr, "<b>Notes:</b> [R.medical.notes]")
-					to_chat(usr, "<a href='?src=\ref[src];medrecordComment=`'>\[View Comment Log\]</a>")
+					var/message = "<b>Medical information</b>\n\n" \
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:704]:
```suggestion
					var/message = "<b>Medical Records: [R.name]</b>\n\n" \
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 16]
```diff
@@ -614,16 +619,20 @@
 			var/datum/record/general/R = SSrecords.find_record("name", perpname)
 			if(istype(R) && istype(R.security))
 				if(hasHUD(usr, "security"))
+					var/message = "<b>Comments: [name]</b>\n\n"
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:622]:
```suggestion
					var/message = "<b>Security Record Comments: [name]</b>\n\n"
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Review by Geevies — APPROVED** [review]
the examine blocks with those borders look really cool

**Pull request merged**